okhttp/src/main/java/okhttp3/internal/ws/CompressionCorrector.kt

```diff
@@ -0,0 +1,49 @@
+package okhttp3.internal.ws
+
+import okio.Buffer
+import okio.ByteString.Companion.decodeHex
+
+/**
+ * Applies post-deflate and pre-inflate buffer transformations
+ * described in [rfc7692#section-7.2](https://tools.ietf.org/html/rfc7692#section-7.2)
+ */
+object CompressionCorrector {
```

Comment:
I would suggest to make it a `class` to avoid having an unnecessary singleton implementation (which will be in memory all the time). Plus this class is stateless and doesn’t seem to need instance in the first place.

Reply:
Agree on the unnecessary `INSTANCE`. 
As this class is essentially 2 stateless functions, I've removed the `object` in favor of top level functions.